tests/includes/class-test-dispatcher.php

```diff
@@ -344,4 +344,95 @@ public function add_follower( $pre, $actor ) {
 			'endpoints'         => array( 'sharedInbox' => 'https://example.org/sharedInbox' ),
 		);
 	}
+
+	/**
+	 * Test that in_reply_to URLs from the same domain are ignored.
+	 *
+	 * @covers ::add_inboxes_of_replied_urls
+	 */
+	public function test_ignore_same_domain_in_reply_to() {
+		// Create a test activity with in_reply_to pointing to same domain.
+		$activity = new Activity();
+		$activity->set_type( 'Create' );
+		$activity->set_id( 'https://example.com/test-id' );
+		$activity->set_in_reply_to( 'https://example.com/post/123' );
+
+		// Create a test actor.
+		$actor_id = self::$user_id;
+
+		// Get inboxes for the activity.
+		$inboxes = Dispatcher::add_inboxes_of_replied_urls( array(), $actor_id, $activity );
+
+		// Verify that no inboxes were added since the in_reply_to is from same domain.
+		$this->assertEmpty( $inboxes, 'Inboxes should be empty for same domain in_reply_to URLs' );
+	}
+
+	/**
+	 * Test that in_reply_to URLs from different domains are processed.
+	 *
+	 * @covers ::add_inboxes_of_replied_urls
+	 */
+	public function test_process_different_domain_in_reply_to() {
+		// Create a test activity with in_reply_to pointing to different domain.
+		$activity = new Activity();
+		$activity->set_type( 'Create' );
+		$activity->set_id( 'https://example.com/test-id' );
+		$activity->set_in_reply_to( 'https://mastodon.social/@user/123456789' );
+
+		// Create a test actor.
+		$actor_id = self::$user_id;
+
+		// Mock the HTTP response for the remote object.
+		add_filter(
+			'pre_http_request',
+			function ( $pre, $parsed_args, $url ) {
+				if ( 'https://mastodon.social/@user/123456789' === $url ) {
+					return array(
+						'response' => array( 'code' => 200 ),
+						'body'     => \wp_json_encode(
+							array(
+								'type'         => 'Note',
+								'id'           => 'https://mastodon.social/@user/123456789',
+								'attributedTo' => 'https://mastodon.social/@user',
+							)
+						),
+					);
+				}
+				return $pre;
+			},
+			10,
+			3
+		);
+
+		// Mock the HTTP response for the actor.
+		add_filter(
+			'pre_http_request',
+			function ( $pre, $parsed_args, $url ) {
+				if ( 'https://mastodon.social/@user' === $url ) {
+					return array(
+						'response' => array( 'code' => 200 ),
+						'body'     => \wp_json_encode(
+							array(
+								'type'  => 'Person',
+								'id'    => 'https://mastodon.social/@user',
+								'inbox' => 'https://mastodon.social/inbox',
+							)
+						),
+					);
+				}
+				return $pre;
+			},
+			10,
+			3
+		);
+
+		// Get inboxes for the activity.
+		$inboxes = Dispatcher::add_inboxes_of_replied_urls( array(), $actor_id, $activity );
+
+		// Verify that the inbox was added.
+		$this->assertContains( 'https://mastodon.social/inbox', $inboxes, 'Inbox should be added for different domain in_reply_to URLs' );
+
+		// Clean up.
+		remove_all_filters( 'pre_http_request' );
```

Comment:
I worry this would remove the `http_disable_request` callback in `bootstrap.php`. 
Alternatively the callbacks in this function could be saved to a variable and then removed individually.